tests/mocha/connection_test.js

```diff
@@ -784,4 +784,348 @@ suite('Connection', function() {
       });
     });
   });
+
+  suite('getPlaceForOrphanedOutput', function() {
+    setup(function() {
+      this.workspace = new Blockly.Workspace();
+
+      Blockly.defineBlocksWithJsonArray([
+        {
+          'type': 'input',
+          'message0': '%1',
+          'args0': [
+            {
+              'type': 'input_value',
+              'name': 'INPUT',
+              'check': 'check'
+            }
+          ],
+        },
+        {
+          'type': 'output',
+          'message0': '',
+          'output': 'check',
+        },
+      ]);
+    });
+
+    teardown(function() {
+      workspaceTeardown.call(this, this.workspace);
+      delete Blockly.Blocks['input'];
+      delete Blockly.Blocks['output'];
```

Comment:
These deletes are actually taken care of in the `sharedTestTeardown` call.
How it works is that the `defineBlocksWithJsonArray` is stubbed in `sharedTestSetup` to keep track of the block definitions that need deleted.

Same comment for the other definition deletes that were created through `defineBlocksWithJsonArray`.

Reply:
Stubbing `defineBlocksWithJsonArray` to handle that is so smart! Deletions removed :D